src/immer/atomWithImmer.ts

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable import/named */
+import { produce, Draft } from 'immer'
+import { atom, WritableAtom } from 'jotai'
+
+export const atomWithImmer = <Value>(initialValue: Value) => {
+  const anAtom: WritableAtom<
+    Value,
+    Function | ((draft: Draft<Value>) => void)
+  > = atom(initialValue, (get, set, fn) =>
+    set(anAtom, produce(get(anAtom), (draft) => fn(draft)) as Function)
```

Comment:
When I tried in codesandbox, it seemed to be working without `as Function`. Not sure. ts version could be different. Not super important either.

Reply:
> @Aslemammad Try adding it in devDependencies and see how it goes.

It's OK now, but wait, I should add the last updates for [better types](https://github.com/pmndrs/jotai/pull/120#discussion_r502401629) .

Reply:
https://github.com/pmndrs/jotai/pull/120#discussion_r502401629
For this one, `as unknown` or `as any` is unavoidable. But, `as Function` looks good. Let's keep it.